File: GameService/src/main/java/com/example/GameService/dto/game/GameStateResponse.java
package com.example.GameService.dto.game;

import com.example.GameService.dto.CardResponse;
import com.example.GameService.enums.AttackType;
import com.example.GameService.enums.CardSuit;
import com.example.GameService.enums.CardType;
import com.example.GameService.enums.GameStatus;

import java.util.List;

public record GameStateResponse(
        List<CardResponse> myCards,
        int myCardsCount,
        List<PlayerGameState> opponents,
        List<CardResponse> table,
        long tableCardsCount,
        long deckCardsCount,
        GameStatus status,
        Long currentPlayerId,
        AttackType attackType,
        Integer attackAmount,
        CardType requestedRank,
        CardSuit requestedSuit,
        Long drawnCardId
) {}

File: frontend/src/app/models/game-state-response.ts
import { CardResponse, CardSuit, CardType } from './card-response';
import { PlayerGameState } from './player-game-state';

export type GameStatus = 'WAITING' | 'IN_PROGRESS' | 'FINISHED';
export type AttackType = 'NONE' | 'TWO' | 'THREE' | 'FOUR' | 'KING_HEARTS' | 'KING_SPADES';

export interface GameStateResponse {
  myCards: CardResponse[];
  myCardsCount: number;
  opponents: PlayerGameState[];
  table: CardResponse[];
  tableCardsCount: number;
  deckCardsCount: number;
  status: GameStatus;
  currentPlayerId: number;
  attackType: AttackType;
  attackAmount: number;
  requestedRank: CardType | null;
  requestedSuit: CardSuit | null;
  drawnCardId: number | null;
}
